challtools/codes.yml

```
@@ -24,6 +24,21 @@ A005:
   level: 3
   formatted_message: 'The following regex flag is missing anchors (''^'', ''$'') in the challenge configuration file (challenge.yml): "{flag}".'
   docs_message: A regex flag missing anchors ('^', '$') n the challenge configuration file (challenge.yml)
+A006:
+  name: Duplicate custom_service_types types
+  level: 4
+  formatted_message: 'The following custom service type contains a duplicate type (challenge.yml): "{type}".'
+  docs_message: A custom service type contains a duplicate type (challenge.yml)
+A007:
+  name: Missing predefined_service display format option
+  level: 4
+  formatted_message: 'The following predefined_service is missing a display format option (challenge.yml): {service}: "{option}".'
+  docs_message: A predefined_service is missing a display format option (challenge.yml)
+A008:
+  name: Missing service type
+  level: 4
+  formatted_message: 'The following predefined_service is referencing missing a service type (challenge.yml): "{service_type}".'
```

Small grammar correction

```suggestion
  formatted_message: 'The following predefined_service is referencing a missing service type (challenge.yml): "{service_type}".'
```